airflow-core/tests/unit/utils/test_helpers.py

```diff
@@ -256,7 +258,7 @@ class MockJobRunner(BaseJobRunner):
     def __init__(self, job: Job, func=None):
         super().__init__(job)
         self.job = job
-        self.job.job_type = self.job_type
+        self.job.job_type = cast("Column", self.job_type)
```

Comment:
Not sure this is the right change, we should probably update `airflow-core/src/airflow/jobs/job.py`. I'll do it